pkg/operator/ceph/object/health.go

```diff
@@ -45,28 +47,30 @@ type bucketChecker struct {
 	context         *clusterd.Context
 	objContext      *Context
 	interval        *time.Duration
-	serviceIP       string
-	port            string
+	port            int32
 	client          client.Client
 	namespacedName  types.NamespacedName
 	healthCheckSpec *cephv1.BucketHealthCheckSpec
+	tlsEnabled      bool
+	tlsCertSecret   string
 }
 
 // newbucketChecker creates a new HealthChecker object
-func newBucketChecker(context *clusterd.Context, objContext *Context, serviceIP, port string, client client.Client, namespacedName types.NamespacedName, healthCheckSpec *cephv1.BucketHealthCheckSpec) *bucketChecker {
+func newBucketChecker(context *clusterd.Context, objContext *Context, port int32, client client.Client, namespacedName types.NamespacedName, objectStoreSpec *cephv1.ObjectStoreSpec) *bucketChecker {
 	c := &bucketChecker{
 		context:         context,
 		objContext:      objContext,
 		interval:        &defaultHealthCheckInterval,
-		serviceIP:       serviceIP,
 		port:            port,
 		namespacedName:  namespacedName,
 		client:          client,
-		healthCheckSpec: healthCheckSpec,
+		healthCheckSpec: &objectStoreSpec.HealthCheck,
+		tlsEnabled:      objectStoreSpec.IsTLSEnabled(),
```

Comment:
Instead of adding tlsEnabled and tlsCertSecret, let's pass the `objectStoreSpec`, then access that spec directly inside the bucket checker. Then we don't need to keep adding new fields to the struct when we can just have them on the `objectStoreSpec`.